app/assets/javascripts/pretty-text/engines/discourse-markdown/mentions.js.es6

```diff
@@ -20,14 +20,40 @@ function addMention(buffer, matches, state) {
+        .source;
+      return new RegExp(
+        `@((?:[_]|${alnum})(?:[._-]|${alnum}){0,58}(?:${alnum}))|@(?:([_]|${alnum}))`
+      );
+    }
+  } else {
+    return /@(\w[\w.-]{0,58}[^\W_])|@(\w)/;
```

Comment:
The old regex didn't detect `@username_` as a valid mention of `@username` even though usernames can't end with an underscore.

Reply:
One question here, markdown.it has some stuff here, https://github.com/markdown-it/uc.micro#readme not sure if it is usable or not.

Reply:
Unfortunately those regular expression aren't usable in our case.